openpower/patches/vesnin-patches/skiboot/skiboot-1001-core-direct-controls-Add-sreset-support-for-P8.patch

```diff
@@ -0,0 +1,52 @@
+From 81b30b55eea07d84a4e9d84e73495a5a67f83ecf Mon Sep 17 00:00:00 2001
+From: Maxim Polyakov <m.polyakov@yadro.com>
+Date: Fri, 18 Jan 2019 17:49:37 +0300
+Subject: [PATCH] core/direct-controls: Add sreset support for P8
+
+VESNIN server uses NMI SRESET signal to initiate memory dumping with
+kdump in case of crash of the host OS. It is not necessary to return
+control to the host's kernel since the Linux has been already
+crashed. This commit permits registration and handling
+opal_signal_system_reset signal for POWER8 based systems.
+
+Signed-off-by: Maxim Polyakov <m.polyakov@yadro.com>
+---
+ core/direct-controls.c | 12 +++++++-----
+ 1 file changed, 7 insertions(+), 5 deletions(-)
+
+diff --git a/core/direct-controls.c b/core/direct-controls.c
+index 04b93a1..b05e6bc 100644
+--- a/core/direct-controls.c
++++ b/core/direct-controls.c
+@@ -824,7 +824,7 @@ int64_t opal_signal_system_reset(int cpu_nr)
+ 	struct cpu_thread *cpu;
+ 	int64_t ret;
+ 
+-	if (proc_gen != proc_gen_p9)
++	if (proc_gen != proc_gen_p9 && proc_gen != proc_gen_p8)
+ 		return OPAL_UNSUPPORTED;
+ 
+ 	/*
+@@ -856,13 +856,15 @@ void direct_controls_init(void)
+ 	if (chip_quirk(QUIRK_MAMBO_CALLOUTS))
+ 		return;
+ 
+-	if (proc_gen != proc_gen_p9)
++	if (proc_gen != proc_gen_p9 && proc_gen != proc_gen_p8)
```

Comment:
I would use a macro like `#define SRESET_SUPPORTED(proc_gen)` because you repeat the same compound conditional twice.

Reply:
I dont' agree, with SRESET_SUPPORTED macro we have to update additional lines with same check but they are not related to this patch.

Reply:
I want to make this fix in the next patch.